### PR TITLE
Add V2 support

### PR DESCRIPTION
**Pull request by ddriddle**
Plan is to drop the daemon, and use an external script for rotating the credentials.

Things left TODO:

+ [x] Fix failure to overwrite existing keys when setting `credential_process` in `~/.aws/credentials` during `aws configure` #195.
+ [x] Update documentation for V2 (See [here](https://github.com/wbingli/awscli-plugin-endpoint/blob/master/README.md) for an example)
+ [x] Add Python 3.11 to testing matrix
+ [x] Drop `awscli` dependency
+ [x] Test with awscli v2 (See [Upgrade docs](https://docs.aws.amazon.com/cli/latest/userguide/cliv2-migration-changes.html#cliv2-migration-profile-plugins))
+ [x] Split awscli_login into two or three submodules.
   + [x] Need a submodule that contains the plugins `login`, `logout`, and `configure`
+ [x] Add error handler to `aws_get_credentials`
+ [x] Rename `aws_get_credentials` to `aws-login-credentials`
+ [x] Add warning to user if keys are set in profile. Warn on login and configure (Should this be in a separate PR? Need to think about this.)
+ [ ] ~Add unit test for `remove_credentials`~
+ [ ] ~Improve integration test for logout to reflect new behavior.~

Documentation:

+ [x] Update README.md documentation (Need to mention V2, need upgrade instructions tested with both V1 and V2)
+ [x] Review `aws logon help`, `aws login configure help`,  and `aws logout help`
+ [x] Update `login`, `configure`, and `logout` online documentation as needed

Testing left TODO:

+ [x] Need unit tests for `raise_if_logged_out`
+ [x] Update integration tests to also test aws v2 (aws v1 tests added here https://github.com/techservicesillinois/awscli-login/pull/120)
  + [x] Add new section in github actions to run aws v2 integration tests. I recommend right before publishing but after unit tests and integration tests for aws v1. Only one run is needed (not per version of Python) since aws v2 bundles its own Python.

Stretch Goals (DO NOT DO IN THIS PR):

+ [ ] Add flag `--all` so that `aws logout --all` clears all credentials & roles in `~/.aws-logon/credentials`.
+ [ ] Add flag to remove cookies to force reauth with Shibboleth. Maybe  `--force` to `aws logout`?
+ [ ] Add error handling to login role selection
+ [ ] Add optional aliases for account numbers
+ [ ] Add default value for ECP value for `aws login configure`. Pull from default or other existing profile.
+ [ ] Tab completion for aws login configure would be awesome. Pull from any available profile.

### CODE REVIEW

**Comment by tzturner** [comment]
End of session today.

Note: using `awscli` V2
``
$ aws login
`No module named 'lxml.etree'
```

**Comment by edthedev** [comment]
We may have hit a bug in `awscli` plugin support.

```sh
PS C:\src\aws-cli> git diff 5cdf69f5c562f1adc07ba95aa033c69dfe5e12f1 d3053cf6429bcffa0507e4aaf411486f61bd89d3 -- .\awscli\plugin.py 
diff --git a/awscli/plugin.py b/awscli/plugin.py
index f993ec1f7..d47128a44 100644
--- a/awscli/plugin.py
+++ b/awscli/plugin.py
@@ -46,6 +46,10 @@ def _import_plugins(plugin_names):
     plugins = []
     for name, path in plugin_names.items():
         log.debug("Importing plugin %s: %s", name, path)
-        if '.' not in name:
+        if '.' not in path:
             plugins.append(__import__(path))
+        else:
+            package, module = path.rsplit('.', 1)
+            module = __import__(path, fromlist=[module])
+            plugins.append(module)
     return plugins
```

Note that this change from `name` to `path` doesn't look intentional, and is why our automated test fails. Not sure at this time why local testing can pass, while testing on CI/CD fails. So this might not be the true reason.

```sh
-        if '.' not in name:
+        if '.' not in path:
```

But the closest open issue I could find is:
https://github.com/aws/aws-cli/issues/2350

**Comment by edthedev** [comment]
We're going to add this to our CI/CD to check if there's an import issue being buried by other layers of code:

```
PYTHONPATH=/home/zdc/dev/projects/awscli-login/venv2/lib/python3.11/site-packages `head -n1 $(which aws) | sed 's/#!//'` -c "print(__import__('awscli_login.plugin', fromlist=['plugin']))"
```

**Comment by edthedev** [comment]
Probably next step: do a manually install of AWS into our CI/CD to test more directly.
https://statics.teams.cdn.office.net/evergreen-assets/safelinks/1/atp-safelinks.html

**Comment by edthedev** [comment]
Team discussion update - to close this out, we will need to pivot away from any reliance on `plugin` mode, since the official support from AWS is not there on V2. Don't anticipate significant usability impact, just need to think through and make the code changes.

- Make use as a plugin completely optional.
- Clarify in README that use in `plugin` mode with AWS CLI V2 is not officially supported.

**Comment by edthedev** [comment]
This pull request - https://github.com/techservicesillinois/awscli-login/pull/149 targets this branch, in order to speed up debugging test issues here.

**Comment by ddriddle** [comment]
Nota bene the config step needs to cleanup empty aws_access_key_id fields otherwise errors are given by aws. For example the following credentials file:

```
[default]
aws_access_key_id = 
aws_secret_access_key = 
aws_session_token = 
aws_security_token = 
credential_process = aws-login --profile default
```

will cause the following error:

```
 aws s3 ls

An error occurred (AuthorizationHeaderMalformed) when calling the ListBuckets operation: The authorization header is malformed; a non-empty Access Key (AKID) must be provided in the credential.
```

Removing the empty fields clears the error.

**Comment by ddriddle** [comment]
Release critical bug: `$ aws login configure` does not properly clear a profile in `~/.aws/credentials` when configuring it. It will leave the security token unchanged and the other values are set to nothing. Leaving the security token causes `aws s3 ls` to fail.

File: docs/readme/body.rst [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+path run the following command::
+
+    $ aws configure set plugins.cli_legacy_plugin_path /usr/lib/python3.12/site-packages
+
+On POSIX systems such as macOS and Linux the preceding can be set
+more easily using the following one-liner::
+
+    $ aws configure set plugins.cli_legacy_plugin_path $(pip show awscli-login | sed -nr 's/^Location: (.*)/\1/p')
+
+If you receive a bad interpreter error or other error please see
```
Comment (mpitcel) [inline docs/readme/body.rst:35]:
```suggestion
If you receive a bad interpreter error, or another error, please see
```

File: docs/readme/body.rst [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+must be supplied as well. This can be looked up using the following
+command::
+
+    $ pip show awscli-login
+    Name: awscli-login
+    Version: 0.2b2.dev64
+    Summary: Plugin for the AWS CLI that retrieves and rotates credentials using SAML ECP and STS.
+    Home-page:
+    Author:
+    Author-email: "David D. Riddle" <ddriddle@illinois.edu>
```
Comment (mpitcel) [inline docs/readme/body.rst:17]:
```suggestion
    Author-email: "Author Name" <author@illinois.edu>
```

**Comment by edthedev** [comment]
During ensemble today, we verified the updated install instructions with a fresh install of AWS CLI V2 on Windows.

**Review by zdc217 — APPROVED** [review]

**Review by mpitcel — APPROVED** [review]

**Pull request merged**